src/Effects/EffectMgr.cpp

```diff
@@ -117,6 +120,149 @@ static const rtosc::Ports local_ports = {
                 }
             }
         }},
+    {"numerator::i", rShort("num") rDefault(0) rLinear(0,99)
+        rProp(parameter) rDoc("Numerator of ratio to bpm"), NULL,
+        [](const char *msg, rtosc::RtData &d)
+        {
+            EffectMgr *eff = (EffectMgr*)d.obj;
+            if(rtosc_narguments(msg)) {
+                int val = rtosc_argument(msg, 0).i;
+                if (val) {
+                    eff->numerator = val;
+                    int Pdelay, Pfreq;
+                    float freq;
+                    switch(eff->nefx) {
+                    case 2:
+                        // invert:
+                        // delay = (Pdelay / 127.0f * 1.5f); //0 .. 1.5 sec
+                        Pdelay = (int)roundf((20320.0f / (float)eff->time->tempo) * 
+                                             ((float)eff->numerator / (float)eff->denominator));
+                        if (eff->numerator&&eff->denominator)
+                            eff->seteffectparrt(2, Pdelay);
+                        break;
+                    case 3:
+                    case 4:
```

Comment:
To make things more clear, it would be great to have a comment hinting what the various cases refer to (i.e. the subtype of effect)

Reply:
good idea.